### PR TITLE
Show blake2b hash of unrecognized operations

### PR DESCRIPTION
**Pull request by 3noch**

### CODE REVIEW

**Comment by 3noch** [comment]
Outstanding question:
  * The `base58_encoded_string` is not used.
  * Should we use hex encoding of hash or base58 encoding? It currently uses hex encoding but that's rather long (64 characters!). base58 is shorter but more complex, of course.

File: src/apdu_sign.c [hunk 1]
```diff
@@ -510,18 +530,35 @@ size_t handle_apdu_sign(uint8_t instruction) {
...
 
-    os_memmove(G.message_data + G.message_data_length, dataBuffer, dataLength);
+    memmove(G.message_data + G.message_data_length, dataBuffer, dataLength);
     G.message_data_length += dataLength;
 
-    if (!last) {
+    if (last) {
+#     ifndef BAKING_APP
+        if (instruction == INS_SIGN) {
+            blake2b_incremental_hash(
```
Comment (3noch) [inline src/apdu_sign.c]: Why must/do we call this before finishing the hash?

Comment (jhartzell42) [reply src/apdu_sign.c]: Hmm, I mean, I'm pretty sure it wouldn't work if we left this call out, weird as it seems.

**Comment by 3noch** [comment]
@jhartzell42 Ok this now shows base58 encoding instead. It also *actually works* with the baking app (it didn't before). It's a pretty large refactor though, so I can go over it with you.

**Pull request merged**